### PR TITLE
BPA Application Workflow changes

### PR DESCRIPTION
**Pull request by sivaete**
BPA Application Workflow changes, code clean up and bug fixes.

Please note, we had to change workflow container component logic for BPA purpose.

### CODE REVIEW

File: web/rainmaker/dev-packages/egov-bpa-dev/src/ui-config/screens/specs/egov-bpa/search-preview.js [hunk 1]
```diff
@@ -132,10 +132,12 @@ const prepareDocumentsView = async (state, dispatch) => {
             .slice(13)
         )) ||
       `Document - ${index + 1}`;
-    return doc;
+      console.log('doooc', doc)
```
Comment (tarunlalwani) [inline web/rainmaker/dev-packages/egov-bpa-dev/src/ui-config/screens/specs/egov-bpa/search-preview.js]: Remove this

File: web/rainmaker/dev-packages/egov-bpa-dev/src/ui-config/screens/specs/egov-bpa/search.js [hunk 2]
```diff
@@ -43,6 +43,7 @@ const pageResetAndChange = (state, dispatch) => {
 
 const startApplyFlow = (state, dispatch) => {
   dispatch(prepareFinalObject("BPAs", []));
+  dispatch(prepareFinalObject("srutinyDetails", {}));
```
Comment (tarunlalwani) [inline web/rainmaker/dev-packages/egov-bpa-dev/src/ui-config/screens/specs/egov-bpa/search.js]: `srutinyDetails`, typo in spelling, please correct. It should `scrutinyDetails`

File: web/rainmaker/dev-packages/egov-bpa-dev/src/ui-config/screens/specs/egov-bpa/search.js [hunk 3]
```diff
@@ -145,7 +146,7 @@ const NOCSearchAndResult = {
                   }
                 },
 
-                buttonLabel: getLabel({
+                buttonLabfireNOCNumberel: getLabel({
```
Comment (tarunlalwani) [inline web/rainmaker/dev-packages/egov-bpa-dev/src/ui-config/screens/specs/egov-bpa/search.js]: what is the `el` at the end?

File: web/rainmaker/dev-packages/egov-bpa-dev/src/ui-config/screens/specs/egov-bpa/searchResource/searchResults.js [hunk 4]
```diff
@@ -170,42 +156,16 @@ export const searchResults = {
 };
 
 const onRowClick = rowData => {
-  switch (rowData[5]) {
+  switch (rowData[3]) {
```
Comment (tarunlalwani) [inline web/rainmaker/dev-packages/egov-bpa-dev/src/ui-config/screens/specs/egov-bpa/searchResource/searchResults.js]: Please define constants for the index with proper label and then use them. Fixes indexes access is bad

File: web/rainmaker/dev-packages/egov-bpa-dev/src/ui-utils/commons.js [hunk 5]
```diff
@@ -135,12 +132,35 @@ export const createUpdateBpaApplication = async (state, dispatch, status) => {
       state.screenConfiguration.preparedFinalObject,
       "BPAs[0].BPADetails.plotdetails.citytown"
     );
-
+    const address = {
```
Comment (tarunlalwani) [inline web/rainmaker/dev-packages/egov-bpa-dev/src/ui-utils/commons.js]: Why is this hardcoded?

File: web/rainmaker/dev-packages/egov-bpa-dev/src/ui-utils/commons.js [hunk 6]
```diff
@@ -165,9 +185,38 @@ export const createUpdateBpaApplication = async (state, dispatch, status) => {
...
     // });
 
-    let documents = [
-      {"documentType": "OWNER.IDENTITYPROOF.VOTERID"}
-    ];
+    let documents;
+    let wfDocuments;
+    if(method === 'UPDATE'){
+      documents = payload.documents;
+      //hard coding these values but time being untill we fix documents capture issue.
```
Comment (tarunlalwani) [inline web/rainmaker/dev-packages/egov-bpa-dev/src/ui-utils/commons.js]: Make this `//TODO: ` so it is easier to locate later

File: web/rainmaker/dev-packages/egov-bpa-dev/src/ui-config/screens/specs/egov-bpa/search-preview.js (hunk 1 above)
Comment (sivaete) [reply web/rainmaker/dev-packages/egov-bpa-dev/src/ui-config/screens/specs/egov-bpa/search-preview.js]: Removed

File: web/rainmaker/dev-packages/egov-bpa-dev/src/ui-config/screens/specs/egov-bpa/search.js (hunk 3 above)
Comment (sivaete) [reply web/rainmaker/dev-packages/egov-bpa-dev/src/ui-config/screens/specs/egov-bpa/search.js]: changed to buttonLabel

File: web/rainmaker/dev-packages/egov-bpa-dev/src/ui-config/screens/specs/egov-bpa/searchResource/searchResults.js (hunk 4 above)
Comment (sivaete) [reply web/rainmaker/dev-packages/egov-bpa-dev/src/ui-config/screens/specs/egov-bpa/searchResource/searchResults.js]: done.

File: web/rainmaker/dev-packages/egov-bpa-dev/src/ui-utils/commons.js (hunk 6 above)
Comment (sivaete) [reply web/rainmaker/dev-packages/egov-bpa-dev/src/ui-utils/commons.js]: done.

**Pull request merged**